www/js/controllers.js

```diff
@@ -1,56 +1,131 @@
 angular.module('homechef.controllers', [])
 
-.controller('AppCtrl', function($scope, $ionicModal, $timeout) {
-
-  // With the new view caching in Ionic, Controllers are only called
-  // when they are recreated or on app start, instead of every page change.
-  // To listen for when this page is active (for example, to refresh data),
-  // listen for the $ionicView.enter event:
-  //$scope.$on('$ionicView.enter', function(e) {
-  //});
-
-  // Form data for the login modal
-  $scope.loginData = {};
-
-  // Create the login modal that we will use later
-  $ionicModal.fromTemplateUrl('templates/login.html', {
-    scope: $scope
-  }).then(function(modal) {
-    $scope.modal = modal;
-  });
+  .controller('AppCtrl', function ($scope, $ionicModal, $ionicLoading, $timeout, $rootScope, $auth, $ionicPopup) {
+
+    // With the new view caching in Ionic, Controllers are only called
+    // when they are recreated or on app start, instead of every page change.
+    // To listen for when this page is active (for example, to refresh data),
+    // listen for the $ionicView.enter event:
+    //$scope.$on('$ionicView.enter', function(e) {
+    //});
+
+    // Form data for the login modal
+    $scope.loginData = {};
+
+    $scope.signupData = {};
+
+    // Create the login modal that we will use later
+    $ionicModal.fromTemplateUrl('templates/login.html', {
+      scope: $scope
+    }).then(function (modal) {
+      $scope.loginmodal = modal;
+    });
+
+    $ionicModal.fromTemplateUrl('templates/signup.html', {
+      scope: $scope
+    }).then(function (modal) {
+      $scope.signupmodal = modal;
+    });
+
+    // Triggered in the login modal to close it
+    $scope.closeLogin = function () {
+      $scope.loginmodal.hide();
+    };
+
+    $scope.closeSignup = function () {
+      $scope.signupmodal.hide();
+    };
+
+    // Open the login modal
+    $scope.login = function () {
+      $scope.loginmodal.show();
+    };
 
-  // Triggered in the login modal to close it
-  $scope.closeLogin = function() {
-    $scope.modal.hide();
-  };
+    $scope.signup = function () {
+      $scope.signupmodal.show();
+    };
 
-  // Open the login modal
-  $scope.login = function() {
-    $scope.modal.show();
-  };
+    $scope.doSignup = function () {
+      $ionicLoading.show({
+        template: 'Signing up...'
+      });
+      $auth.submitRegistration($scope.signupData)
+        .then(function (resp) {
+          $ionicLoading.hide();
+          $scope.closeSignup();
+          $ionicLoading.show({
+            template: 'Successfully signed up!',
+            duration: 1000
+          });
+        })
+        .catch(function (response) {
+          $ionicLoading.hide();
+          $scope.errorMessage = response;
+          $scope.showAlert('Failure', response.data.errors.full_messages);
```

Comment:
`errors.full_messages` is an Array. I would like to see a screenshot of what this looks like if there are more than one error